components/insight/SRC/org/openmicroscopy/shoola/util/roi/io/ROIReader.java

```diff
@@ -207,10 +220,37 @@ private void formatShape(Roi roi, ShapeData shape)
         if (roi.getFillColor() != null) {
             settings.setFill(roi.getFillColor());
         }
-
+        int pos = roi.getPosition();
         int c = roi.getCPosition();
         int z = roi.getZPosition();
         int t = roi.getTPosition();
+
+        ImagePlus image = roi.getImage();
+        int imageC = image.getNChannels();
+        int imageT = image.getNFrames();
+        int imageZ = image.getNSlices();
+        if (imageC == 1 && imageZ == 1) {
+            shape.setC(0);
+            shape.setZ(0);
+            z = 0;
+            c = 0;
+            t = pos;
+        } else if (imageZ == 1 && imageT == 1) {
+            c = pos;
+            z = 0;
+            t = 0;
+            shape.setZ(0);
+            shape.setT(0);
+        } else if (imageC == 1 && imageT == 1) {
+            z = pos;
+            t = 0;
+            c = 0;
+            shape.setC(0);
+            shape.setT(0);
+        }
+        if (c >= imageC || z >= imageZ || t >= imageT) {
```

Comment:
@jburel: This check will leave out ROIs on the last frame, since imageC, Z, and T are 1-based (while c, z, and t are 0-based). I'd suggest to change it to `if (c > imageC || z > imageZ || t > imageT)`.